Additional updates to PRR docs (#3)
lachie83/enhancements: keps/sig-network/563-dual-stack/README.md

* **Does enabling the feature change any default behavior?**
  Pods and Services will remain single-stack until cli flags have been modified
  as described in this KEP. Existing and new services will remain single-stack
  until user requests otherwise. Pods will become dual-stack once CNI is
  configured for dual-stack.

* **Can the feature be disabled once it has been enabled (i.e. can we roll back
  the enablement)?**
  Yes. If you decide to turn off dual-stack after turning on:
    1. Ensure all services are converted to single-stack first (downgraded to
       single-stack as described in this KEP)
    2. Remove the CLI parameters.

>     3. Disable the feature.

Reviewer: Are these two discrete steps?

Author: Yes; Kal stipulated that they should be in that order.

  This feature will not be operable if either kube-apiserver or etcd is unavailable.

* **What are other known failure modes?**

  * Missing prerequisites. Operator must verify the following conditions:
    1. Ensure correct support in the node infrastructure provider.
      a. supports routing both IPv4 and IPv6 interfaces.
      b. makes both IPv4 and IPv6 interfaces available to Kubernetes.

>     2. CNI needs to be correctly configured for dual-stack service.
>       a. Pods need to be able to use dual-stack service.

Reviewer: ```suggestion
    2. CNI provider needs to be correctly configured for dual-stack.
      a. Kubernetes must be able to assign IPv4 and IPv6 IP addresses from the CNI provider.
```

>     3. Service cidrblock for controller manager must be available.

Reviewer: ```suggestion
    3. Dual-stack CLI flags must be configured on the cluster as defined in the [docs](https://kubernetes.io/docs/concepts/services-networking/dual-stack/#enable-ipv4-ipv6-dual-stack)
```

          where applicable.
    4. Operator can ensure that `endpoints` and `endpointSlices` are correctly 
       created for the service in question by using kubectl.
    5. If the pod is using host network then operator must ensure that the node
       is correctly reporting dual-stack addresses.
    6. Due to the amount of time needed for control loops to function, when
       scaling with dual-stack it may take time to attach all ready endpoints.

  * CNI changes may affect legacy workloads.
    1. When dual-stack is configured and enabled, DNS queries will start returning
       IPv4(A) and IPv6(AAAA).
    2. If a workload doesn't account for being offered both IP families, it
       may fail in unexpected ways. For example, firewall rules may need to be
       updated to allow IPv6 addresses.
    3. Recommended to independently verify legacy workloads to ensure fidelity.

  *  IP-related error conditions to consider.
    1.  pod IP allocation fails (this is due to CNI)
      a. Will result in the pod not running if there is no IP allocated from
         the CIDR.

>     2. routing fails

Reviewer: ```suggestion
    2. Service to pod routing fails
```
